x-pack/legacy/plugins/siem/public/components/url_state/helpers.ts

```diff
@@ -129,7 +129,8 @@ export const isKqlForRoute = (
     (currentLocation === CONSTANTS.hostsPage && queryLocation === CONSTANTS.hostsPage) ||
     (currentLocation === CONSTANTS.networkPage && queryLocation === CONSTANTS.networkPage) ||
     (currentLocation === CONSTANTS.hostsDetails && queryLocation === CONSTANTS.hostsDetails) ||
-    (currentLocation === CONSTANTS.networkDetails && queryLocation === CONSTANTS.networkDetails)
+    (currentLocation === CONSTANTS.networkDetails && queryLocation === CONSTANTS.networkDetails) ||
+    (currentLocation === CONSTANTS.detectionsPage && queryLocation === CONSTANTS.detectionsPage)
```

Comment:
FYI -- it looks like `isKqlForRoute` has no usages outside of tests. Is this 💀code?

Reply:
Let me check